pages/desktop/user.py

```diff
@@ -119,8 +124,28 @@ def click_update_account(self):
         self.selenium.find_element(*self._update_account_locator).click()
         WebDriverWait(self.selenium, self.timeout).until(lambda s: self.update_message == "Profile Updated")
 
-    def change_hide_email_state(self):
-        self.selenium.find_element(*self._hide_email_checkbox).click()
+    def hide_email_address(self):
+        email_checkbox = self.selenium.find_element(*self._email_checkbox)
+        WebDriverWait(self.selenium, self.timeout).until(expected_conditions.visibility_of(email_checkbox));
+
+        if email_checkbox.is_selected():
+            err_msg = 'Email is already hidden'
+            return err_msg
+
+        if not email_checkbox.is_selected():
+            self.selenium.find_element(*self._email_checkbox).click()
+        return email_checkbox.is_selected() == True
+
+    def show_email_address(self):
```

Comment:
The same changes as above need to be applied to this method to, i.e.,
1. Find the checkbox element.
2. Wait for it to be visible.
3. If it is already unchecked, raise an exception.
4. Click it (which should uncheck it).
5. Wait for it to be unchecked.